lib/src/rules/rule.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:meta/meta.dart';
+
+import '../models/issue.dart';
+import '../models/rule_documentation.dart';
+import '../models/severity.dart';
+
+/// Interface that code checker uses to communicate with the rules
```

Comment:
I think, `code checker` mention should be removed. Also, change to `An interface`.